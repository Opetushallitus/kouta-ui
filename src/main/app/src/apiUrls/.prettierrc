{
  "printWidth": 1000,
  "trailingComma": "all",
  "tabWidth": 2,
  "semi": true,
  "singleQuote": true
}