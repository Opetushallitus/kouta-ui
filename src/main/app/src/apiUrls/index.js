import { development as developmentUrls } from './urls';

export const configure = async urls => {
  const { NODE_ENV, REACT_APP_CYPRESS } = process.env;

  const isCypress = !!REACT_APP_CYPRESS;

  if (['development', 'test'].includes(NODE_ENV)) {
    urls.addProperties(developmentUrls({ isCypress }));
  } else {
    await urls.load({ overrides: '/kouta/rest/config/frontProperties' });
  }

  return urls;
};

export default configure;
