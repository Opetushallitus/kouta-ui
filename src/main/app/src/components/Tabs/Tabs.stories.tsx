import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Tabs, { Tab } from './index';

storiesOf('Tabs', module).add('Basic', () => (
  <div style={{ borderBottom: '1px solid #999999' }}>
    <Tabs value="b" onChange={action('change')}>
      <Tab value="a">Tab A</Tab>
      <Tab value="b">Tab B</Tab>
      <Tab value="c">Tab C</Tab>
    </Tabs>
  </div>
));
