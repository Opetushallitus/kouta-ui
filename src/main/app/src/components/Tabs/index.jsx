import React from 'react';
import styled, { css } from 'styled-components';

import { getThemeProp } from '../../theme';

const Wrapper = styled.div`
  display: flex;
`;

const activeCss = css`
  border: 1px solid ${getThemeProp('palette.border')};
  border-top: 2px solid ${getThemeProp('palette.primary.main')};
  border-bottom: 1px solid white;
  color: ${getThemeProp('palette.text.primary')};
`;

export const Tab = styled.div`
  padding: 10px 20px;
  position: relative;
  z-index: 1;
  bottom: -1px;
  font-family: ${getThemeProp('typography.fontFamily')};
  font-size: 1rem;
  color: ${getThemeProp('palette.primary.main')};
  border-bottom: 0px none;
  margin-right: 6px;
  cursor: pointer;
  border: 1px solid transparent;
  border-top: 2px solid transparent;

  ${({ active }) => active && activeCss};
`;

const Tabs = ({ value, onChange = () => {}, ...props }) => {
  const children = React.Children.map(props.children, child => {
    const active = value !== undefined && child.props.value === value;

    return React.cloneElement(child, {
      active,
      onClick: () => {
        onChange(child.props.value);
      },
    });
  });

  return <Wrapper {...props}>{children}</Wrapper>;
};

export default Tabs;
