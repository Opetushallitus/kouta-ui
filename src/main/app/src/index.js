import 'core-js/features/object/keys';
import 'core-js/features/object/values';
import 'core-js/features/promise';
import 'core-js/features/array/includes';
import 'core-js/features/array/every';
import 'core-js/features/array/flat-map';

import React from 'react';
import ReactDOM from 'react-dom';
import createBrowserHistory from 'history/createBrowserHistory';
import { urls as ophUrls } from 'oph-urls-js';
import get from 'lodash/get';
import merge from 'lodash/merge';

import * as serviceWorker from './serviceWorker';
import App from './components/App';
import createStore from './state';
import defaultTheme from './theme';
import configureUrls from './apiUrls';
import createLocalisation from './localisation';
import { getLocalisation } from './apiUtils';
import getTranslations from './translations';
import createHttpClient from './httpClient';

const history = createBrowserHistory({ basename: 'kouta' });

serviceWorker.unregister();

const isDev = process.env.NODE_ENV === 'development';

const loadLocalisation = async ({
  namespace,
  language,
  httpClient,
  apiUrls,
}) => {
  const localisation = await getLocalisation({
    category: namespace,
    locale: language,
    httpClient,
    apiUrls,
  });

  const translations = getTranslations();

  return get(translations, [language, namespace])
    ? merge({}, translations[language][namespace], localisation || {})
    : localisation;
};

(async () => {
  const apiUrls = await configureUrls(ophUrls);

  const httpClient = createHttpClient({
    apiUrls,
    callerId: !isDev ? process.env.REACT_APP_CALLER_ID : null,
  });

  const localisationInstance = createLocalisation({
    debug: isDev,
    loadLocalisation: ({ namespace, language }) =>
      loadLocalisation({ namespace, language, httpClient, apiUrls }),
  });

  window.__i18n__ = localisationInstance;

  const { store, persistor } = createStore({
    apiUrls,
    httpClient,
    history,
    localisation: localisationInstance,
  });

  window.__store__ = store;

  ReactDOM.render(
    <App
      store={store}
      theme={defaultTheme}
      urls={apiUrls}
      httpClient={httpClient}
      history={history}
      localisation={localisationInstance}
      persistor={persistor}
    />,
    document.getElementById('root'),
  );
})();
