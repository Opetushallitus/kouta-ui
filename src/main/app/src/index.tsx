import React from 'react';

import { Globals } from '@react-spring/web';
import { inspect } from '@xstate/inspect';
import { createBrowserHistory } from 'history';
import { urls as ophUrls } from 'oph-urls-js';
import ReactDOM from 'react-dom';

import createHttpClient from './httpClient';
import { createDefaultLocalization } from './localization';
import App from './pages/App';
import { createStore } from './state';
import defaultTheme from './theme';
import { configure as configureUrls } from './urls';
import { isPlaywright } from './utils';

Globals.assign({
  skipAnimation: isPlaywright,
});

if (import.meta.env.VITE_XSTATE_INSPECTOR) {
  inspect({
    iframe: false, // open in new window
  });
}

const history = createBrowserHistory({ basename: 'kouta' });

(async () => {
  let apiUrls = ophUrls;

  const httpClient = createHttpClient({
    apiUrls,
    callerId: import.meta.env.VITE_CALLER_ID,
  });

  apiUrls = await configureUrls(apiUrls, httpClient);

  const localizationInstance = await createDefaultLocalization({
    httpClient,
    apiUrls,
  });

  const { store, persistor } = createStore();

  ReactDOM.render(
    <App
      store={store}
      theme={defaultTheme}
      urls={apiUrls}
      httpClient={httpClient}
      history={history}
      localization={localizationInstance}
      persistor={persistor}
    />,
    document.getElementById('root')
  );
})();
