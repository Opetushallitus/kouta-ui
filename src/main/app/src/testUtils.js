import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import defaultTheme from './theme';

const makeFnWithTheme = fn => tree => {
  return fn(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>);
};

export const shallowWithTheme = makeFnWithTheme(shallow);

export const mountWithTheme = makeFnWithTheme(mount);

export const setMockDate = date => {
  const _Date = Date;

  global.Date.now = () => +date;

  return () => {
    global.Date = _Date;
  };
};
