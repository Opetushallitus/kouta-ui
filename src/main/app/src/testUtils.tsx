import React, { PropsWithChildren } from 'react';

import { configureStore } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';

import { createRootReducer } from './state/rootReducer';
import defaultTheme from './theme';

const makeFnWithTheme = fn => tree => {
  return fn(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>);
};

export const renderWithTheme = makeFnWithTheme(render);

export const mountWithTheme = tree => {
  const { container } = makeFnWithTheme(render)(tree);
  return container.firstChild;
};

export const setMockDate = date => {
  const _Date = Date;

  global.Date.now = () => Number(date);

  return () => {
    global.Date = _Date;
  };
};

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({ reducer: createRootReducer({}), preloadedState }),
    ...renderOptions
  }
) {
  // eslint-disable-next-line @typescript-eslint/ban-types
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return (
      <Provider store={store}>
        <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
      </Provider>
    );
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';
