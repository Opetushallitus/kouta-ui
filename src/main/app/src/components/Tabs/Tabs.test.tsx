import React from 'react';

import Tabs, { Tab } from './index';
import { mountWithTheme } from '#/src/testUtils';

test('renders correctly', () => {
  expect(
    mountWithTheme(
      <Tabs value="b">
        <Tab value="a">Tab A</Tab>
        <Tab value="b">Tab B</Tab>
        <Tab value="c">Tab C</Tab>
      </Tabs>
    )
  ).toMatchSnapshot();
});
