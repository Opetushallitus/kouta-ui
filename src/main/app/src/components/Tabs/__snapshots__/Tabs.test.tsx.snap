// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 {
  outline: none;
  background-color: transparent;
  border: 0px none;
  cursor: pointer;
  -webkit-transition: color 0.25s,border-color 0.25s;
  transition: color 0.25s,border-color 0.25s;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-bottom: 3px solid transparent;
  padding: 6px 20px;
  line-height: 1.5;
  font-family: 'Roboto',sans-serif;
  font-size: 1rem;
  color: #666666;
  font-weight: 500;
}

.c1:hover,
.c1active {
  color: #159ecb;
}

.c1:focus {
  border-color: rgba(21,158,203,0.3);
}

.c2 {
  outline: none;
  background-color: transparent;
  border: 0px none;
  cursor: pointer;
  -webkit-transition: color 0.25s,border-color 0.25s;
  transition: color 0.25s,border-color 0.25s;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-bottom: 3px solid transparent;
  padding: 6px 20px;
  line-height: 1.5;
  font-family: 'Roboto',sans-serif;
  font-size: 1rem;
  color: #666666;
  font-weight: 500;
  color: #159ecb;
}

.c2:hover,
.c2active {
  color: #159ecb;
}

.c2:focus {
  border-color: rgba(21,158,203,0.3);
}

.c2,
.c2:focus {
  border-color: #159ecb;
}

<div
  className="c0"
  role="tablist"
>
  <button
    className="c1"
    disabled={false}
    onClick={[Function]}
    role="tab"
    value="a"
  >
    Tab A
  </button>
  <button
    className="c2"
    disabled={false}
    onClick={[Function]}
    role="tab"
    value="b"
  >
    Tab B
  </button>
  <button
    className="c1"
    disabled={false}
    onClick={[Function]}
    role="tab"
    value="c"
  >
    Tab C
  </button>
</div>
`;
